src/lib.rs

```diff
@@ -61,7 +62,7 @@ use num_traits::Zero;
 ///     }
 /// }
 /// ```
-pub trait Metric<P> {
+pub trait Metric<P>: Default {
```

Comment:
I think the nearest neighbor data structure should implement default only if the metric also implements default. I don't think we should put this bound here.

Reply:
If we don't have this bound, then we should derive Default for most metric types, since `KnnBatchInsert` depends on KNN types implementing default, which depends on metric types implementing default.

Reply:
> If we don't have this bound, then we should derive Default for most metric types, since `KnnBatchInsert` depends on KNN types implementing default, which depends on metric types implementing default.

I agree with this. This will have to be up to downstream implementations to derive these traits where possible.